zrml/global-disputes/src/types.rs

```diff
@@ -2,9 +2,88 @@ use frame_support::pallet_prelude::{Decode, Encode, MaxEncodedLen, TypeInfo};
 use zeitgeist_primitives::types::OutcomeReport;
 
+#[derive(Debug, TypeInfo, Decode, Encode, MaxEncodedLen, Clone, PartialEq, Eq)]
+pub enum Possession<AccountId, Balance, Owners> {
+    Paid { owner: AccountId, fee: Balance },
+    Shared { owners: Owners },
+}
+
+impl<AccountId, Balance, Owners> Possession<AccountId, Balance, Owners> {
+    pub fn is_shared(&self) -> bool {
+        matches!(self, Possession::Shared { .. })
+    }
+
+    pub fn get_shared_owners(self) -> Option<Owners> {
+        match self {
+            Possession::Shared { owners } => Some(owners),
+            _ => None,
+        }
+    }
+
+    pub fn is_paid(&self) -> bool {
+        matches!(self, Possession::Paid { .. })
+    }
+}
+
+/// The information about a voting outcome of a global dispute.
+#[derive(Debug, TypeInfo, Decode, Encode, MaxEncodedLen, Clone, PartialEq, Eq)]
+pub struct OutcomeInfo<AccountId, Balance, Owners> {
+    /// The current sum of all locks on this outcome.
+    pub outcome_sum: Balance,
+    /// The information about the owner(s) and optionally additional fee.
+    pub possession: Option<Possession<AccountId, Balance, Owners>>,
+}
+
+/// The general information about the global dispute.
+#[derive(Debug, TypeInfo, Decode, Encode, MaxEncodedLen, Clone, PartialEq, Eq)]
+pub struct GDInfo<AccountId, Balance, Owners, BlockNumber> {
+    /// The outcome, which is in the lead.
+    pub winner_outcome: OutcomeReport,
+    /// The information about the winning outcome.
+    pub outcome_info: OutcomeInfo<AccountId, Balance, Owners>,
+    /// The current status of the global dispute.
+    pub status: GDStatus<BlockNumber>,
+}
+
+impl<AccountId, Balance: Saturating, Owners: Default, BlockNumber>
+    GDInfo<AccountId, Balance, Owners, BlockNumber>
+{
+    pub fn new(outcome: OutcomeReport, vote_sum: Balance) -> Self {
+        let outcome_info = OutcomeInfo { outcome_sum: vote_sum, possession: None };
+        GDInfo { winner_outcome: outcome, status: GDStatus::Initialized, outcome_info }
+    }
+
+    pub fn update_winner(&mut self, outcome: OutcomeReport, vote_sum: Balance) {
+        self.winner_outcome = outcome;
+        self.outcome_info.outcome_sum = vote_sum;
+    }
+}
+
+#[derive(TypeInfo, Debug, Decode, Encode, MaxEncodedLen, Clone, PartialEq, Eq)]
+pub enum GDStatus<BlockNumber> {
+    /// The global dispute is initialized.
+    Initialized,
+    /// The global dispute is in progress. Save the addition of outcome end and vote end.
+    /// The block number `add_outcome_end`, when the addition of new outcomes is over.
+    /// The block number `vote_end`, when the global dispute voting period is over.
+    Active { add_outcome_end: BlockNumber, vote_end: BlockNumber },
+    /// The global dispute is finished.
+    Finished,
+    /// The global dispute was triggered to get destroyed.
+    Destroyed,
+}
+
+pub struct RewardInfo<MarketId, AccountId, Balance> {
+    pub market_id: MarketId,
+    pub reward: Balance,
+    pub source: AccountId,
+}
```

Comment:
I see your point. We can remove it. I just want to say before: Using structs instead of function parameters is a way to reduce the number of parameter for a given function. So assume we would have even more fields, then a struct would speak for itself instead of passing all parameters to a function.

Reply:
f67e92bad518fb8136d557e71dc5c4644c13ffad

Reply:
> I just want to say before: Using structs instead of function parameters is a way to reduce the number of parameter for a given function.

Yeah, absolutely. In some situations, this can be a real advantage. I think in this case, you just structure the parameters before the call and then destructure them after the call. I agree that this is pretty nitpicky, but I think you're not gaining anything by doing this here.